cbpfc.go

```diff
@@ -601,7 +601,15 @@ func guaranteedGuard(targets map[pos]*block, opts packetGuardOpts) packetGuard {
 			continue
 		}
 
-		if guaranteed := guaranteedGuard(target.jumps, opts); guaranteed > insnsGuard {
+		var guaranteed packetGuard
```

Comment:
I think we can make the `cache` check earlier, right after calling `blockNeverMatches`. Eg:

```
...
		// Block can't match the packet, ignore it
		if blockNeverMatches(target) {
			continue
		}

		// We've already traversed this block
		if guard, ok := seen[target]; ok {
			targetGuards = append(targetGuards, guard)
			continue
		}

		insnsCovered, insnsGuard := opts.requiredGuard(target.insns)
...
```